service/src/main/java/bio/terra/landingzone/stairway/flight/delete/DeleteLandingZoneFlight.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.landingzone.stairway.flight.delete;
+
+import bio.terra.landingzone.common.utils.LandingZoneFlightBeanBag;
+import bio.terra.landingzone.common.utils.RetryRules;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.RetryRule;
+import bio.terra.stairway.Step;
+
+public class DeleteLandingZoneFlight extends Flight {
+  @Override
+  public void addStep(Step step, RetryRule retryRule) {
+    super.addStep(step, retryRule);
+  }
+
+  /**
+   * All subclasses must provide a constructor with this signature.
+   *
+   * @param inputParameters FlightMap of the inputs for the flight
+   * @param applicationContext Anonymous context meaningful to the application using Stairway
+   */
+  public DeleteLandingZoneFlight(FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+    final LandingZoneFlightBeanBag flightBeanBag =
+        LandingZoneFlightBeanBag.getFromObject(applicationContext);
+
+    addDeleteSteps(flightBeanBag);
+  }
+
+  private void addDeleteSteps(LandingZoneFlightBeanBag flightBeanBag) {
```

Comment:
There should probably be a step to delete the Sam resource (e.g. [this method](https://github.com/DataBiosphere/terra-landing-zone-service/blob/main/service/src/main/java/bio/terra/landingzone/service/iam/LandingZoneSamService.java#L163))

Reply:
Sounds good, I will add that.